numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1783,36 +1792,44 @@ unpack_bits(PyObject *input, int axis)
         unpack_init = 1;
     }
 
-    NPY_BEGIN_THREADS_THRESHOLDED(PyArray_DIM(new, axis));
-
-    n_in = PyArray_DIM(new, axis);
+    in_n = count / 8;
+    in_tail = count % 8;
     in_stride = PyArray_STRIDE(new, axis);
     out_stride = PyArray_STRIDE(out, axis);
 
-    while (PyArray_ITER_NOTDONE(it)) {
+    NPY_BEGIN_THREADS_THRESHOLDED(in_n);
+
+    while (PyArray_ITER_NOTDONE(ot)) {
         npy_intp index;
         unsigned const char *inptr = PyArray_ITER_DATA(it);
         char *outptr = PyArray_ITER_DATA(ot);
+        unsigned char mask;
```

Comment:
Why did this move? Better to leave things in the narrowest scope possible

Reply:
Yes. Forgot to move it back when I realized I could optimize the stride=1 loop without using the mask.